docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,328 @@
+
+We use the following example of a non-nested bundle `Parent` to let us state all of the alignment relationships between members of `p`.
+
+```scala mdoc:silent
+import chisel3._
+class Parent extends Bundle {
+  val foo = UInt(32.W)
+  val bar = Flipped(UInt(32.W))
+}
+class MyModule0 extends Module {
+  val p = Wire(new Parent)
+}
+```
+
+First, every member is always aligned with themselves:
+ - `p` is aligned w.r.t `p`
+ - `p.foo` is aligned w.r.t `p.foo`
+ - `p.bar` is aligned w.r.t `p.bar`
+
+Next, we list all parent/child relationships. Because the `bar` field is `Flipped`, it changes its aligment relative to its parent. 
+ - `p` is aligned w.r.t `p.foo`
+ - `p` is flipped w.r.t `p.bar`
+
+Finally, we can list all sibling relationships:
+ - `p.foo` is flipped w.r.t `p.bar`
+
+The next example has a nested bundle `GrandParent` who instantiates an aligned `Parent` field and flipped `Parent` field.
+
+```scala mdoc:silent
+import chisel3._
+class GrandParent extends Bundle {
+  val parent = new Parent()
+  val flippedParent = Flipped(new Parent())
+}
+class MyModule1 extends Module {
+  val g = Wire(new GrandParent)
+}
+```
+
+Consider the following alignements between grandparent and grandchildren. An odd number of flips indicate a flipped relationship; even numbers of flips indicate an aligned relationship.
+ - `g` is aligned w.r.t `g.flippedParent.bar`
+ - `g` is aligned w.r.t `g.parent.foo`
+ - `g` is flipped w.r.t `g.flippedParent.foo`
+ - `g` is flipped w.r.t `g.parent.bar`
+
+Consider the following alignment relationships starting from `g.parent` and `g.flippedParent`. *Note that whether `g.parent` is aligned/flipped relative to `g` has no effect on the aligned/flipped relationship between `g.parent` and `g.parent.foo` because alignment is only relative to the two members in question!*:
+ - `g.parent` is aligned w.r.t. `g.parent.foo`
+ - `g.flippedParent` is aligned w.r.t. `g.flippedParent.foo`
+ - `g.parent` is flipped w.r.t. `g.parent.bar`
+ - `g.flippedParent` is flipped w.r.t. `g.flippedParent.bar`
+
+In summary, a field is aligned or flipped w.r.t. another member of the hardware component. This means that the type of the consumer/producer is the only information needed to determine the behavior of any operator. *Whether the consumer/producer is a subfield of a larger bundle is irrelevant; you ONLY need to know the type of the consumer/producer*.
+
+## Input/Output/IO (Maybe talk about later once I fix Chisel directionality?)
+
+`Input(gen)`/`Output(gen)` are coercing operators. They perform two functions: (1) create a new Chisel type that has all flips removed from all recursive children fields but structurally equivalent to `gen`, and (2) apply `Flipped` if `Input`, keep aligned (do nothing) if `Output`.
+
```

Comment:
we could put a note to link to the issue about fixing this stuff

Reply:
val foo = IO(Input(Bool())) val bar = IO(Output(Bool)))

foo is aligned to foo
bar is aligned to bar
foo :<>= bar would attempt to generate foo <= bar, chisel is fine w that but we check that foo is not assignable ("can firrtl connect aggregates")

Reply:
val foo = IO (Flipped(Decoupled(Bool)) val bar = IO (Aligned(Decoupled(Bool))

foo.bits :<>= bar.bits
// foo.bits is aligned relative to foo.bits, bar.bits is aligned relative to bar.bits

foo.bits  <= bar.bits